### PR TITLE
Event handlers

### PR DESCRIPTION
**Pull request by mikethreels**
### When updating the project the following points have been added.
- [x] Changed App to a stateful component.
- [x] implemented the handleClick function.
- [x] implemented the onClick event.

apologies for the lack of commits.

### CODE REVIEW

File: src/components/button_panel.jsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      <div className="group1">
+        <Button key="AC" name="buttongroup1" clickHandler={handleClick} buttonName="AC" />
+        <Button key="+/-" name="buttongroup1" clickHandler={handleClick} buttonName="+/-" />
+        <Button key="%" name="buttongroup1" clickHandler={handleClick} buttonName="%" />
+        <Button key="÷" name="buttongroup1last" clickHandler={handleClick} buttonName="÷" />
+      </div>
+      <div className="group2">
+        <Button key="8" name="buttongroup2" clickHandler={handleClick} buttonName="8" />
+        <Button key="9" name="buttongroup2" clickHandler={handleClick} buttonName="9" />
+        <Button key="7" name="buttongroup2" clickHandler={handleClick} buttonName="7" />
```
Comment (kubilaycaglayan) [inline src/components/button_panel.jsx]: Let's correct the number order. 😊

**Review by kubilaycaglayan — CHANGES_REQUESTED** [review]
#### Hello Team 👋
#### Status: Changes Required ♻️

Great job so far! 👏
Click handling is great 👍
Let's cover a few details together to make your project better 😊⏬

### Button Alignment

- [x] On the top row of number buttons, please fix the order of the numbers. Currently, it goes like 8, 9, 7.

### Behaviour

- [x] 160 / 2 = and then % throws a prop type warning on the console.

image

- [x] Consider the edge case, add error handling when the user tries to divide a number by zero. Currently, the application is breaking. 12 / 0 =

## That's it, you can submit another code review request after making these changes ♻️😇

With all my respect 🙏
> Kubilay Caglayan | Technical Support Engineer

**Comment by mikethreels** [comment]
## Milestone: Calculations Update

### When updating the project the following points have been added.
- [x] fixed the order of the buttons.
- [x] fixed the type error when using the % button.
- [x] covered the edge case when divided by 0.
- [x] make sure the divided by 0 error is cleared.

**Review by Torres-ssf — APPROVED** [review]
### Milestone Approved :+1: :clap: 
Hi @mikethreels :hand: 
Awesome job implementing the requested changes :clap: 
Good luck with the next project.

Happy Coding :computer: 
**Sergio Torres**

**Pull request merged**